llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9386,10 +9397,8 @@ void LoopVectorizationPlanner::adjustRecipesForReductions(
                cast<VPInstruction>(&U)->getOpcode() ==
                    VPInstruction::ComputeReductionResult;
       });
-      if (PreferPredicatedReductionSelect ||
-          TTI.preferPredicatedReductionSelect(
-              PhiR->getRecurrenceDescriptor().getOpcode(), PhiTy,
-              TargetTransformInfo::ReductionFlags()))
+      if (CM.usePredicatedReductionSelect(
+              PhiR->getRecurrenceDescriptor().getOpcode(), PhiTy))
```

Comment:
Would it be sufficient to adjust the reduction phi recipe when introducing EVL recipes instead?

Reply:
It can be adjusted in the EVL transformation. patch 8a3982fa6c65c08271b43194698d27e4cb29d0f4
But I don't recommend this. Such an implementation is more complicated and may lose the opportunity to sink the non-predicated reduction select outside the loop in the future VPlan transformation.
Could you point out why you want to adjust it in the EVL transformation?

Reply:
At the moment, everything EVL related is applied during the transform that introduces EVL recipes; one potential issue is that we assume EVL is used here, but the transform may not apply.

Don't have any strong preferences, doing it later indeed seems to require some extra work

Reply:
Sounds good. to me